File: UCA-4/Arrays/C++/array.cpp
/*
* Program For Arrays IUDT(Insertion Updation Deletion Traversal)
*/

#include<iostream>
using namespace std;
int size = 0;

void insertion(int *arr,int pos,int value){
	if(size==pos){
		return;
	}
	arr[pos] = value;
}

void traversal(int *arr){
	printf("Printing Of Array\n");
	for (int i = 0; i < size; ++i)
	{
		printf("%d\n",arr[i]);
	}
}

void updation(int *arr,int pos,int value){
	if(pos>=size || pos<0){
		return;
	}
	arr[pos] = value;
}

void deletion(int *arr,int pos){
	if(pos>=size || pos<0){
		return;
	}
	arr[pos] = 0;
}


int main(int argc, char const *argv[])
{
	size = atoi(argv[1]);
	int arr[size];
	int top = -1;
	int test = size;
	while(test--){
		insertion(arr,++top,atoi(argv[top+2]));
	}
	traversal(arr);
	updation(arr,2,69);
	deletion(arr,4);
	traversal(arr);
	return 0;
}

File: UCA-4/Arrays/Java/array.java
/*
* Program For Arrays IUDT(Insertion Updation Deletion Traversal)
Time Complexities 
Insertion - O(1)
Updateion - O(1)
Deletion - O(1)
Traversal - O(N)
*/
import java.util.*;

class Test {
	/*
	Static Methods Used Here so that it can be called from Main Method
	*/
	static void insert(int[] arr,int pos,int value){
		arr[pos] = value;
	}

	static void traverse(int[] arr){
		System.out.println("Printing Array");
		for (int x:arr) {
			System.out.println(x);
		}
	}

	static void update(int[] arr,int pos,int value){
		arr[pos] = value;
	}

	static void delete(int[] arr,int pos){
		for (int i=pos;i<arr.length-1;i++) {
			arr[i] = arr[i+1];
		}
		arr[arr.length-1] = 0;
	}

	public static void main(String[] args) {
		int size = Integer.parseInt(args[0]);
		int arr[] = new int[size];
		for (int i=0;i<size;i++) {
			insert(arr,i,Integer.parseInt(args[i+1]));
		}
		traverse(arr);
		update(arr,2,69);
		delete(arr,3);
		traverse(arr);
	}
}